libraries/preferences/api/src/main/kotlin/io/element/android/features/preferences/api/store/SessionPreferencesStore.kt

```diff
@@ -19,8 +19,20 @@ package io.element.android.features.preferences.api.store
 import kotlinx.coroutines.flow.Flow
 
 interface SessionPreferencesStore {
+    suspend fun setSharePresence(enabled: Boolean)
+    fun isSharePresenceEnabled(): Flow<Boolean>
+
     suspend fun setSendPublicReadReceipts(enabled: Boolean)
     fun isSendPublicReadReceiptsEnabled(): Flow<Boolean>
 
+    suspend fun setRenderReadReceipts(enabled: Boolean)
+    fun isRenderReadReceiptsEnabled(): Flow<Boolean>
+
+    suspend fun setSendTypingNotifications(enabled: Boolean)
+    fun isSendTypingNotificationsEnabled(): Flow<Boolean>
+
+    suspend fun setRenderTypingNotifications(enabled: Boolean)
+    fun isRenderTypingNotificationsEnabled(): Flow<Boolean>
+
```

Comment:
Is there a reason why we're keeping these separate? When we enabled/disable them we do it to all of them at the same time, so they act as a single one. Or are we thinking about splitting it in the future?

Reply:
It may have sense to invoke separate `set` method, as soon as `setSharePresence` is not invoked anymore.